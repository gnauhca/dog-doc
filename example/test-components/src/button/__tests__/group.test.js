import React from 'react';
import { mount, render, shallow } from 'enzyme';

describe('Button Group', () => {
  it('renders correctly', () => {
    const Demo = require('../demos/group').default;
    const wrapper = mount(<Demo></Demo>);
    expect(wrapper).toMatchSnapshot();
  });
});
