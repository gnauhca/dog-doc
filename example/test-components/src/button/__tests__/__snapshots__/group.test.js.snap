// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Group renders correctly 1`] = `
<_default>
  <div
    className="demo-button-group-base"
  >
    <div>
      <ButtonGroup>
        <div
          className="ten-button-group"
        >
          <Button
            autoWidth={false}
            block={false}
            disabled={false}
            loading={false}
            round={false}
            size="default"
            theme="default"
          >
            <button
              className="ten-button ten-button--size-default ten-button--type-default"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span>
                按钮1
              </span>
            </button>
          </Button>
          <Button
            autoWidth={false}
            block={false}
            disabled={false}
            loading={false}
            round={false}
            size="default"
            theme="default"
          >
            <button
              className="ten-button ten-button--size-default ten-button--type-default"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span>
                按钮2
              </span>
            </button>
          </Button>
          <Button
            autoWidth={false}
            block={false}
            disabled={false}
            loading={false}
            round={false}
            size="default"
            theme="default"
          >
            <button
              className="ten-button ten-button--size-default ten-button--type-default"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span>
                按钮3
              </span>
            </button>
          </Button>
        </div>
      </ButtonGroup>
      <ButtonGroup>
        <div
          className="ten-button-group"
        >
          <Button
            autoWidth={false}
            block={false}
            disabled={false}
            loading={false}
            round={false}
            size="default"
            theme="primary"
          >
            <button
              className="ten-button ten-button--size-default ten-button--type-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span>
                按钮1
              </span>
            </button>
          </Button>
          <Button
            autoWidth={false}
            block={false}
            disabled={false}
            loading={false}
            round={false}
            size="default"
            theme="default"
          >
            <button
              className="ten-button ten-button--size-default ten-button--type-default"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span>
                按钮2
              </span>
            </button>
          </Button>
          <Button
            autoWidth={false}
            block={false}
            disabled={false}
            loading={false}
            round={false}
            size="default"
            theme="default"
          >
            <button
              className="ten-button ten-button--size-default ten-button--type-default"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span>
                按钮3
              </span>
            </button>
          </Button>
        </div>
      </ButtonGroup>
    </div>
    <div>
      <ButtonGroup
        size="large"
      >
        <div
          className="ten-button-group"
        >
          <Button
            autoWidth={false}
            block={false}
            disabled={false}
            loading={false}
            round={false}
            size="default"
            theme="primary"
          >
            <button
              className="ten-button ten-button--size-large ten-button--type-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span>
                按钮1
              </span>
            </button>
          </Button>
          <Button
            autoWidth={false}
            block={false}
            disabled={false}
            loading={false}
            round={false}
            size="default"
            theme="default"
          >
            <button
              className="ten-button ten-button--size-large ten-button--type-default"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span>
                按钮2
              </span>
            </button>
          </Button>
          <Button
            autoWidth={false}
            block={false}
            disabled={false}
            loading={false}
            round={false}
            size="default"
            theme="default"
          >
            <button
              className="ten-button ten-button--size-large ten-button--type-default"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span>
                按钮3
              </span>
            </button>
          </Button>
        </div>
      </ButtonGroup>
      <ButtonGroup
        disabled={true}
      >
        <div
          className="ten-button-group"
        >
          <Button
            autoWidth={false}
            block={false}
            disabled={false}
            loading={false}
            round={false}
            size="default"
            theme="primary"
          >
            <button
              className="ten-button ten-button--disabled ten-button--size-default ten-button--type-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span>
                按钮1
              </span>
            </button>
          </Button>
          <Button
            autoWidth={false}
            block={false}
            disabled={false}
            loading={false}
            round={false}
            size="default"
            theme="default"
          >
            <button
              className="ten-button ten-button--disabled ten-button--size-default ten-button--type-default"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span>
                按钮2
              </span>
            </button>
          </Button>
          <Button
            autoWidth={false}
            block={false}
            disabled={false}
            loading={false}
            round={false}
            size="default"
            theme="default"
          >
            <button
              className="ten-button ten-button--disabled ten-button--size-default ten-button--type-default"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span>
                按钮3
              </span>
            </button>
          </Button>
        </div>
      </ButtonGroup>
    </div>
    <div>
      <ButtonGroup
        block={true}
        style={
          Object {
            "width": 400,
          }
        }
      >
        <div
          className="ten-button-group"
          style={
            Object {
              "width": 400,
            }
          }
        >
          <Button
            autoWidth={false}
            block={false}
            disabled={false}
            loading={false}
            round={false}
            size="default"
            theme="primary"
          >
            <button
              className="ten-button ten-button--block ten-button--size-default ten-button--type-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span>
                按钮1
              </span>
            </button>
          </Button>
          <Button
            autoWidth={false}
            block={false}
            disabled={false}
            loading={false}
            round={false}
            size="default"
            theme="default"
          >
            <button
              className="ten-button ten-button--block ten-button--size-default ten-button--type-default"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span>
                按钮2
              </span>
            </button>
          </Button>
          <Button
            autoWidth={false}
            block={false}
            disabled={false}
            loading={false}
            round={false}
            size="default"
            theme="default"
          >
            <button
              className="ten-button ten-button--block ten-button--size-default ten-button--type-default"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span>
                按钮3
              </span>
            </button>
          </Button>
        </div>
      </ButtonGroup>
    </div>
  </div>
</_default>
`;
